src/engine/sample.h

```diff
@@ -26,39 +31,75 @@ enum DivResampleFilters {
   DIV_RESAMPLE_CUBIC,
   DIV_RESAMPLE_BLEP,
   DIV_RESAMPLE_SINC,
-  DIV_RESAMPLE_BEST
+  DIV_RESAMPLE_BEST,
+  DIV_RESAMPLE_MAX // for identify boundary
+};
+
+enum DivSampleDepth: unsigned char {
+  DIV_SAMPLE_DEPTH_1BIT=0,
+  DIV_SAMPLE_DEPTH_1BIT_DPCM=1,
+  DIV_SAMPLE_DEPTH_QSOUND_ADPCM=4,
+  DIV_SAMPLE_DEPTH_ADPCM_A=5,
+  DIV_SAMPLE_DEPTH_ADPCM_B=6,
+  DIV_SAMPLE_DEPTH_X68K_ADPCM=7,
+  DIV_SAMPLE_DEPTH_8BIT=8,
+  DIV_SAMPLE_DEPTH_BRR=9,
+  DIV_SAMPLE_DEPTH_VOX=10,
+  DIV_SAMPLE_DEPTH_16BIT=16,
+  DIV_SAMPLE_DEPTH_MAX // for identify boundary
+};
+
+enum DivSampleLoopMode: unsigned char {
```

Comment:
Sadly I won't support this on VGM export...

Reply:
that's for support these loop modes natively in chip itself. (ex: ES5506, SCSP, YMF271(maybe?), ICS2115)

Reply:
Yup, I know.